consensus/vbft/chain_store.go

```diff
@@ -24,22 +24,46 @@ import (
 	"github.com/ontio/ontology/common"
 	"github.com/ontio/ontology/common/log"
 	"github.com/ontio/ontology/core/ledger"
+	"github.com/ontio/ontology/core/store"
+	"github.com/ontio/ontology/core/store/overlaydb"
+	"github.com/ontio/ontology/events/message"
 )
 
+type PendingBlock struct {
+	block      *Block
+	execResult *store.ExecuteResult
+}
 type ChainStore struct {
 	db              *ledger.Ledger
 	chainedBlockNum uint32
-	pendingBlocks   map[uint32]*Block
+	pendingBlocks   map[uint32]*PendingBlock
+	server          *Server
+	needSubmitBlock bool
 }
 
-func OpenBlockStore(db *ledger.Ledger) (*ChainStore, error) {
+func OpenBlockStore(db *ledger.Ledger, server *Server) (*ChainStore, error) {
```

Comment:
suggest just adding serverPid to chainstore.

serverPid *actor.PID

Reply:
has fixed